frontend/tests/oauth.test.ts

```diff
@@ -0,0 +1,113 @@
+﻿import {type APIRequestContext, type Browser, expect} from '@playwright/test';
+import {test} from './fixtures';
+import {LoginPage} from './pages/loginPage';
+import {OidcDebuggerPage} from './pages/oidcDebuggerPage';
+import {addUserToProject, loginAs, logout, preApproveOauthApp} from './utils/authHelpers';
+import {OauthApprovalPage} from './pages/oauthApprovalPage';
+import {elawaProjectId, serverBaseUrl} from './envVars';
+
+// we've got a matrix of flows to test
+// pre approved, not approved yet,
+// logged in already, not logged in
+
+//use https://oidcdebugger.com/ to test oauth
+const flows = [
+  {approved: true, loggedIn: false},
+  {approved: false, loggedIn: false},
+  {approved: true, loggedIn: true},
+  {approved: false, loggedIn: true}
+];
+
+test.describe('oauth tests', () => {
+  for (const flow of flows) {
+    const name = `can login with oauth, ${flow.approved ? 'pre-approved' : 'not approved'}, ${flow.loggedIn ? 'logged in' : 'not logged in'}`;
+    test(name, async ({page, tempUser}) => {
+
+      if (flow.approved || flow.loggedIn) await loginAs(page.request, tempUser);
+      if (flow.approved) {
+        await preApproveOauthApp(page.request, OidcDebuggerPage.clientId, OidcDebuggerPage.scopes);
+      }
+      if (!flow.loggedIn) {
+        await logout(page);
+      }
```

Comment:
Could consider putting everything inside the first `if` for clarity, as it does all belong together:
```ts
      if (flow.approved || flow.loggedIn) {
        await loginAs(page.request, tempUser);
        if (flow.approved) {
          await preApproveOauthApp(page.request, OidcDebuggerPage.clientId, OidcDebuggerPage.scopes);
        }
        if (!flow.loggedIn) {
          await logout(page);
        }
     }
```